feature_engine/datetime/datetime_subtraction.py

```diff
@@ -0,0 +1,272 @@
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Add new features.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns
+        -------
+        X_new: Pandas dataframe
+            The input dataframe plus the new variables.
+        """
+
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check that input is a dataframe
+        X = check_X(X)
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_X_matches_training_df(X, self.n_features_in_)
+
+        # reorder variables to match train set
+        X = X[self.feature_names_in_]
+
+        self._sub(X)
+
+        if self.drop_original:
+            X.drop(
+                columns=set(self.variables + self.reference),
+                inplace=True,
+            )
+
+        return X
+
+    def _sub(self, X):
+
+        for reference in self.reference:
+            varname = [f"{var}_sub_{reference}" for var in self.variables]
+            X[varname] = (
+                X[self.variables].sub(X[reference], axis=0)
+                .apply(lambda s: s / np.timedelta64(1, self.output_unit))
+            )
+
+        return X
+
+    def get_feature_names_out(self, input_features: Optional[bool] = None) -> List:
+        """Get output feature names for transformation.
+
+        Parameters
+        ----------
+        input_features: bool, default=None
+            If `input_features` is `None`, then the names of all the variables in the
+            transformed dataset (original + new variables) is returned. Alternatively,
+            if `input_features` is True, only the names for the new features will be
+            returned.
+
+        Returns
+        -------
+        feature_names_out: list
+            The feature names.
+        """
+        check_is_fitted(self)
+
+        if input_features is not None and not isinstance(input_features, bool):
+            raise ValueError(
+                "input_features takes None or a boolean, True or False. "
+                f"Got {input_features} instead."
+            )
+
+        # Names of new features
+        feature_names = []
+        for reference in self.reference:
+            varname = [f"{var}_sub_{reference}" for var in self.variables]
+            feature_names.extend(varname)
+
+        if input_features is None or input_features is False:
+            if self.drop_original is True:
+                # Remove names of variables to drop.
+                original = [
+                    f
+                    for f in self.feature_names_in_
+                    if f not in self.variables + self.reference
+                ]
+                feature_names = original + feature_names
+            else:
+                feature_names = self.feature_names_in_ + feature_names
+
+        return feature_names
```

Comment:
we need to add the tags to the class, this is for the automatic tests. Something like this:

https://github.com/feature-engine/feature_engine/blob/60a70455465be305e9b9e1b3fc17dfbcd1ca2ae8/feature_engine/creation/base_creation.py#L113-L125